### PR TITLE
[MAINTENANCE] Lint Docs

### PR DESCRIPTION
**Pull request by Kilo59**
Enable linting for the docs directory code.
Mostly involved swapping `os.path` and `glob` usage with `pathlib.Path` methods.

Also enabled type-checking for some docs related modules.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by Kilo59** [comment]
@anthonyburdi if I wanted to manually test these changes, what we be the best way to do that?

Or would you expect the CI to fail if something was off?

File: docs/ruff.toml
Comment (Kilo59) [inline docs/ruff.toml:1]: If we want different linting rules for the docs code we can customize it in this file

File: docs/checks/docs_link_checker.py [hunk 1]
```diff
@@ -142,23 +144,24 @@ def _check_external_link(self, link: str, file: str) -> Optional[LinkReport]:
                 link, file, f"External link raised a connection error {err.errno}"
             )
 
-    def _get_os_path(self, path: str) -> str:
-        """Gets an os-specific path from a path found in a markdown file"""
-        return path.replace("/", os.path.sep)
```
Comment (Kilo59) [inline docs/checks/docs_link_checker.py:157]: Don't need this method at all if using `pathlib`

**Review by cdkini — APPROVED** [review]

**Review by anthonyburdi — APPROVED** [review]
LGTM! Thanks for making these updates.

**Review by Shinnnyshinshin — APPROVED** [review]
🙇🏼

**Pull request merged**